eeps/eep-0072.md

```diff
@@ -0,0 +1,227 @@
+    Author: Jesse Gumm <ja(at)gumm(dot)io>
+    Status: Draft
+    Type: Standards Track
+    Created: 08-Oct-2024
+    Post-History:
+****
+EEP 72: Reserved words and Variables as record names, and enhancement to definition syntax
+----
+
+Abstract
+========
+
+This EEP loosens some of the restrictions around record names to make it
+no longer necessary to quote them when they are named with reserved
+words (`#if` vs `#'if'`) or words with capitalized first characters (`#Hello`
+vs `#'Hello'`).
+
+This EEP also proposes to add a new record-like syntax to the record
+definitions (also adopting the above syntactical changes), so that the
+following record definitions would all be valid and identical:
+
+```erlang
+-record('div', {a :: integer(), b :: integer()}).
+-record #'div'{a :: integer(), b :: integer()}.
+-record #div{a :: integer(), b :: integer()}.
+```
+
+The latter two are proposed new syntax.  The following would also be valid
+and identical since parentheses are optional in attributes:
+
+```erlang
+-record 'div', {a :: integer(), b :: integer()}.
+-record(#'div'{a :: integer(), b :: integer()}).
+-record(#div{a :: integer(), b :: integer()}).
+```
+
+Usage Syntax Motivation
+=======================
+
+Record names are atoms. As such, the current Erlang syntax requires the record
+names to be consistent with the rest of the language's use of atoms.
+
+All atoms in Erlang can be denoted with single quotes. Some examples:
+
+For example:
+```erlang
+'foo'.
+'FOO'.
+'foo-bar'.
+```
+
+But, conveniently, simple atoms (all alphanumeric, underscores (`_`) or
+at symbols (`@`) with the first character being a lowercase letter and not one
+of the 20+ reserved words), in all contexts can be invoked without the
+necessary wrapping quotes. Some examples:
+
+```erlang
+foo.
+foo_Bar.
+'foo-bar'. % still quoted since the term has a non-atomic character in it.
+```
+
+Conveniently, this also means that records named with simple atoms can be
+invoked and used without having to quote the atoms. For example:
+
+```erlang
+-record(foo, {a, b}).
+
```

Comment:
I am trying to squeeze in a capital letter record further down, but is not sure this works out well enough...  It is a suggestion.

```suggestion
-record(bar, {c}).

```